### PR TITLE
Provide row indexset when building sparsity pattern

### PR DESCRIPTION
**Pull request by fdrmrc**
Current master is not working  with latest deal.II version because `DynamicSparsityPattern` and `SparsityTools::distribute_sparsity_pattern` need to be given the index set of locally relevant rows. 

While there, I updated the deal.II test file

### CODE REVIEW

**Comment by fdrmrc** [comment]
While there, I updated the deal.II test file

**Comment by fdrmrc** [comment]
@luca-heltai The two apps are not working in parallel, right? 
If you try `mpirun -np 2 laplacian_debug ../prms/one_circle_one_coeff.prm` (also with the master branch) you should hit a runtime error in `distribute_local_to_global()`

**Comment by luca-heltai** [comment]
Well, they did run in parallel. I'm not sure what happened, but at least in the elasticity branch, things run. It may well be that with a single inclusion things don't work in parallel, but in principle this should not be the case. Can you open an issue with the problem you observed?

File: source/elasticity.cc [hunk 1]
```diff
@@ -129,9 +129,12 @@ ElasticityProblem<dim, spacedim>::make_grid()
     {
       GridIn<spacedim> gi;
       gi.attach_triangulation(tria);
-      // std::ifstream infile(par.name_of_grid);
-      const std::string infile(par.name_of_grid);
+#ifdef DEAL_II_WITH_GMSH_API
+      std::string infile(par.name_of_grid);
       Assert(!infile.empty(), ExcIO());
```
Comment (luca-heltai) [inline source/elasticity.cc:134]: This assert does not make sense for a string. The check on the existence of the file is done inside the function below for the gmsh api. The assert should be moved to the other branch of the `#ifdef`

**Review by luca-heltai — APPROVED** [review]

**Comment by fdrmrc** [comment]
> Well, they did run in parallel. I'm not sure what happened, but at least in the elasticity branch, things run. It may well be that with a single inclusion things don't work in parallel, but in principle this should not be the case. Can you open an issue with the problem you observed?


I've opened a PR for that. The elasticity example was already working in the distributed case.

**Pull request merged**